### PR TITLE
feat: db stacks transactions validated incorrectly

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes https://github.com/stacks-network/sbtc/issues/523.

## Changes

* This change makes sure that we only extract stacks transactions that were deployed by the appropriate entity.

## Testing Information

There was a test for extracting relevant stacks transactions and it was updated to check that contract calls to smart contracts with the incorrect stacks addresses are ignored.

## Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: signer/src/stacks/wallet.rs [hunk 1]
```diff
@@ -495,4 +498,42 @@ mod tests {
...
+        let wallet = SignerWallet::new(&pks1, 5, network_kind, 0).unwrap();
+
+        let rotate_keys = RotateKeysTransaction {
+            txid: fake::Faker.fake_with_rng(&mut rng),
+            aggregate_key: wallet.aggregate_key(),
+            signer_set: public_keys,
+            signatures_required: wallet.signatures_required,
+        };
+
+        // Although it's unlikely, it's possible for the shuffle to not
```
Comment (cylewitruk) [inline signer/src/stacks/wallet.rs:529]: Was this something that you had intended to include or is it just here for mental documentation?

File: signer/tests/integration/postgres.rs [hunk 2]
```diff
@@ -97,14 +97,16 @@ async fn should_be_able_to_query_bitcoin_blocks() {
     signer::testing::storage::drop_db(store).await;
 }
 
-struct InitiateWithdrawalRequest;
+struct InitiateWithdrawalRequest {
+    deployer: StacksAddress,
+}
 
 impl AsContractCall for InitiateWithdrawalRequest {
     const CONTRACT_NAME: &'static str = "sbtc-withdrawal";
```
Comment (cylewitruk) [inline signer/tests/integration/postgres.rs:105]: Unrelated to this PR specifically, but we should probably create a `constants.rs` or something in the root where we keep the contract names etc.

**Review by cylewitruk — APPROVED** [review]
Looks good :)

File: signer/src/stacks/wallet.rs (hunk 1 above)
Comment (djordon) [reply signer/src/stacks/wallet.rs:529]: ha, this entire file diff, from https://github.com/stacks-network/sbtc/pull/599/commits/f63ba4fa1b851757c9b37ae01f065325488fe49d) isn't supposed to be here. I was supposed to be in another branch. I'll remove it.

**Pull request merged**